Lib/asyncio/tasks.py

```diff
@@ -430,8 +432,8 @@ async def wait_for(fut, timeout):
         # wait until the future completes or the timeout
         try:
             await waiter
-        except exceptions.CancelledError:
-            if fut.done():
+        except exceptions.CancelledError as e:
+            if fut.done() and _SENTINEL in e.args:
```

Comment:
```suggestion
            if fut.done() and e.args == (_SENTINEL,):
```
or `len(e.args) == 1 and e.args[0] == _SENTINEL

Stricter test is better, isn't it?